src/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php

```diff
@@ -684,82 +695,70 @@ public function processMultiLineArray($phpcsFile, $stackPtr, $arrayStart, $array
             to be moved back one space however, then both errors would be fixed.
         */
 
-        $numValues = count($indices);
-
-        $indicesStart  = ($keywordStart + 1);
-        $indexLine     = $tokens[$stackPtr]['line'];
-        $lastIndexLine = null;
+        $indicesStart = ($keywordStart + 1);
         foreach ($indices as $index) {
-            if ($index['value'] === false) {
+            $valuePointer = $index['value'];
+            if ($valuePointer === false) {
                 // Syntax error or live coding.
                 continue;
             }
 
             if (isset($index['index']) === false) {
                 // Array value only.
-                if ($tokens[$index['value']]['line'] === $tokens[$stackPtr]['line'] && $numValues > 1) {
-                    $error = 'The first value in a multi-value array must be on a new line';
-                    $fix   = $phpcsFile->addFixableError($error, $stackPtr, 'FirstValueNoNewline');
-                    if ($fix === true) {
-                        $phpcsFile->fixer->addNewlineBefore($index['value']);
-                    }
-                }
-
                 continue;
             }
 
-            $lastIndexLine = $indexLine;
-            $indexLine     = $tokens[$index['index']]['line'];
+            $indexPointer = $index['index'];
+            $indexLine    = $tokens[$indexPointer]['line'];
 
             if ($indexLine === $tokens[$stackPtr]['line']) {
                 $error = 'The first index in a multi-value array must be on a new line';
-                $fix   = $phpcsFile->addFixableError($error, $index['index'], 'FirstIndexNoNewline');
+                $fix   = $phpcsFile->addFixableError($error, $indexPointer, 'FirstIndexNoNewline');
                 if ($fix === true) {
-                    $phpcsFile->fixer->addNewlineBefore($index['index']);
+                    $phpcsFile->fixer->addNewlineBefore($indexPointer);
                 }
 
                 continue;
             }
 
-            if ($indexLine === $lastIndexLine) {
+            $previous = $phpcsFile->findPrevious([T_WHITESPACE, T_COMMA], ($indexPointer - 1), ($arrayStart + 1), true);
+            if ($previous !== false && $tokens[$previous]['line'] === $indexLine) {
                 $error = 'Each index in a multi-line array must be on a new line';
-                $fix   = $phpcsFile->addFixableError($error, $index['index'], 'IndexNoNewline');
+                $fix   = $phpcsFile->addFixableError($error, $indexPointer, 'IndexNoNewline');
                 if ($fix === true) {
-                    if ($tokens[($index['index'] - 1)]['code'] === T_WHITESPACE) {
```

Comment:
Seems like both
```
$tokens[($index['index'] - 1)]['code'] === T_WHITESPACE
```
and 
```
$prevToken = $phpcsFile->findPrevious(T_WHITESPACE, ($token - 1), null, true);

for ($i = ($prevToken + 1); $i < $token; $i++) {
    $phpcsFile->fixer->replaceToken($i, '');
}

$phpcsFile->fixer->addNewLineBefore($token);
```
was used...

Reply:
I'm sorry but I don't get your point at all.